xarray/core/groupby.py

```diff
@@ -279,7 +279,7 @@ def _yield_binary_applied(self, func, other):
                 raise TypeError('GroupBy objects only support binary ops '
                                 'when the other argument is a Dataset or '
                                 'DataArray')
-            except KeyError:
+            except (KeyError, ValueError):
```

Comment:
OK, makes sense. Given that we want to catch this, maybe better to make an `IndexingError` exception type that subclasses ValueError to make this more explicitly. There are probably a few existing errors that could be switched to this.